tests/test_celery.py

```diff
@@ -0,0 +1,277 @@
+# SOFTWARE.
+#
+###############################################################################
+import os
+import copy
+from time import sleep
+import aioredis
+from sqlalchemy.inspection import inspect
+
+from utils import GoFishArgs, add_test_jobs, add_test_session
+from tuna.dbBase.sql_alchemy import DbSession
+from tuna.utils.machine_utility import load_machines
+from tuna.miopen.db.tables import MIOpenDBTables
+from tuna.miopen.miopen_lib import MIOpen
+from tuna.miopen.utils.config_type import ConfigType
+from tuna.utils.utility import serialize_job_config_row, arch2targetid
+from tuna.miopen.celery_tuning.celery_tasks import prep_kwargs
+from tuna.machine import Machine
+from tuna.libraries import Operation
+from tuna.celery_app.celery_workers import launch_worker_per_node
+from tuna.celery_app.utility import get_q_name
+from tuna.parse_args import TunaArgs, setup_arg_parser
+from tuna.miopen.celery_tuning.celery_tasks import prep_worker
+from tuna.miopen.worker.fin_utils import compose_config_obj, fin_job
+from tuna.miopen.utils.lib_helper import get_worker
+
+
+def test_celery_workers():
+  miopen = MIOpen()
+  miopen.args = GoFishArgs()
+  miopen.args.label = 'tuna_pytest_celery'
+  miopen.args.session_id = add_test_session(label=miopen.args.label)
+
+  #load jobs
+  dbt = MIOpenDBTables(config_type=ConfigType.convolution)
+  num_jobs = add_test_jobs(miopen, miopen.args.session_id, dbt,
+                           miopen.args.label, miopen.args.label,
+                           ['miopen_perf_compile'],
+                           'test_add_celery_compile_job',
+                           'miopenConvolutionAlgoGEMM')
+  #assert num_jobs
+  num_jobs = 4
+
+  machine_lst = load_machines(miopen.args)
+  machine = machine_lst[0]
+  miopen.operation = Operation.COMPILE
+  miopen.dbt = MIOpenDBTables(session_id=miopen.args.session_id,
+                              config_type=ConfigType.convolution)
+  miopen.args.enqueue_only = False
+  db_name = os.environ['TUNA_DB_NAME']
+
+  #testing get_q_name
+  q_name = get_q_name(miopen, op_compile=True)
+  assert q_name == f"compile_q_{db_name}_sess_{miopen.args.session_id}"
+  q_name = get_q_name(miopen, op_eval=True)
+  assert q_name == f"eval_q_{db_name}_sess_{miopen.args.session_id}"
+
+  #testing prep_tuning
+  _, subp_list = miopen.prep_tuning()
+  assert subp_list
+  for subp in subp_list:
+    print(subp.pid)
+    subp.kill()
+
+  miopen.args.enqueue_only = True
+  _, subp_list = miopen.prep_tuning()
+  assert subp_list == []
+
+
+  cmd = f"celery -A tuna.celery_app.celery_app worker -l info -E -n tuna_HOSTNAME_sess_{miopen.args.session_id} -Q test_{db_name}"  #pylint: disable=line-too-long
+  #testing launch_worker_per_node
+  subp_list = launch_worker_per_node([machine], cmd, True)
+  sleep(2)
+  assert subp_list
+  assert miopen.cancel_consumer(f"test_{db_name}")
+  for subp in subp_list:
+    print(subp.pid)
+    subp.kill()
```

Comment:
cancel_consumer would also kill the workers for the named queue. Perhaps this loop would confirm that the subp were terminated by this call.

Reply:
i'll add